src/Development/IDE/Plugin/Completions/Logic.hs

```diff
@@ -392,19 +396,7 @@ getCompletions ideOpts cc pm pmapping prefixInfo caps withSnippets = do
           to                             'foo :: Int -> String ->    '
                                                               ^
       -}
-      pos =
```

Comment:
We don't need this stuff anymore because of fuzzy position mapping. I had to wade through multiple git commits across repositories to find @bubba as the source of this code. If I understand correctly, it existed so that we could detect type contexts even if the module didn't parse.

Suppose the cursor is at the `|`

```haskell
foo :: Integer -> Integer -> |
```

So the idea was to strip out the trailing "-> ", and then look for the "context" of the position (whether it was in a value/type/import etc.) at the position (`newPositionToOld` would just return `Nothing`).

Now, `fromDelta` will return the original range that was inserted to/replaced, so we can use it instead.